lib/WebDriver/AbstractWebDriver.php

```diff
@@ -122,15 +124,28 @@ protected function curl($requestMethod, $command, $parameters = null, $extraOpti
 
         list($rawResult, $info) = ServiceFactory::getInstance()->getService('service.curl')->execute($requestMethod, $url, $parameters, $extraOptions);
 
+        // According to https://w3c.github.io/webdriver/webdriver-spec.html all 4xx responses are to be considered an error and return plaintext,
+        // while 5xx responses are json encoded
+        if (substr($info['http_code'], 0, 1) === '4') {
+            throw WebDriverException::factory(Exception::CURL_EXEC, 'Webdriver http error: ' . $info['http_code'] . ', payload :' . substr($rawResult, 0, 1000));
```

Comment:
Wrong namespace. Change the occurrences of Exception::CURL_EXEC to WebDriverException::CURL_EXEC